src/main/java/org/cryptomator/common/vaults/VaultListManager.java

```diff
@@ -134,29 +151,70 @@ public static VaultState.Value redetermineVaultState(Vault vault) {
 		VaultState state = vault.stateProperty();
 		VaultState.Value previousState = state.getValue();
 		return switch (previousState) {
-			case LOCKED, NEEDS_MIGRATION, MISSING -> {
+			case LOCKED, NEEDS_MIGRATION, MISSING, VAULT_CONFIG_MISSING, MASTERKEY_MISSING -> {
 				try {
-					var determinedState = determineVaultState(vault.getPath());
+					var determinedState = determineVaultState(vault.getPath(), vault.getVaultSettings());
+					if (determinedState == MASTERKEY_MISSING) {
+						var vaultScheme = vault.getVaultConfigCache().getUnchecked().getKeyId().getScheme();
+						if (KeyLoadingStrategy.isHubVault(vaultScheme)) {
+							determinedState = LOCKED;
+						}
+					}
 					if (determinedState == LOCKED) {
 						vault.getVaultConfigCache().reloadConfig();
 					}
 					state.set(determinedState);
 					yield determinedState;
 				} catch (IOException e) {
-					LOG.warn("Failed to determine vault state for " + vault.getPath(), e);
+					LOG.warn("Failed to determine vault state for {}", vault.getPath(), e);
 					state.set(ERROR);
 					vault.setLastKnownException(e);
 					yield ERROR;
 				}
 			}
-			case ERROR, UNLOCKED, PROCESSING -> previousState;
+			case ERROR -> {
+				try {
+					var determinedState = determineVaultState(vault.getPath(), vault.getVaultSettings());
+					state.set(determinedState);
+					yield determinedState;
+				} catch (IOException e) {
+					LOG.warn("Failed to redetermine vault state for {}", vault.getPath(), e);
+					vault.setLastKnownException(e);
+					yield ERROR;
+				}
+			}
```

Comment:
For what do we need this extra ERROR case anymore? It's basically the same as the above (LOCKED, NEEDS_MIGRATION,...)